src/common/logging/sqlite_logger.rs

```diff
@@ -61,9 +61,14 @@ fn load_events(conn: &mut Connection, search_criteria: &SearchCriteria) -> Resul
         args.push(&v.value);
     }
 
-    if let Some(ref v) = search_criteria.event_type {
-        where_conds.push(make_where_string("event_type", &v.mode)?);
-        args.push(&v.value);
+    if let Some(ref v) = search_criteria.event_types {
+        let mut conditions: Vec<String> = vec!();
```

Comment:
Is not better to use "IN" operator?
i.e. something like this? 
```rust
        let modes : Vec<_> = v.iter().map(|e| e.mode.clone()).collect();
        where_conds.push(format!("event_type IN ({})", modes.join(",")));
        for event in v {
            args.push(&event.value);
        }
```
(And I also prefere collect than mut vec & push :)

Reply:
Ah I see it now, my suggestion is not right. We would need to introduce whole "mode" with "in".

Reply:
I usually use prepared statements for SQL and IN isn't allowed there :-) I thought about it, but it would complicate the code a bit, so I just reused the existing WHERE infrastructure. I'll rewrite it to collect, I didn't use it because the args need for anyway.

Reply:
The collect is a bit problematic because of the error semantics. To match the rest of the function, it should return an error if any of the parameters are wrong. Is there an idiomatic way to do this in Rust? Something like `v.iter().map(|e| make_where_string(&e.mode)?).collect()` (but I can't use the question mark in the map). It could be rewritten to check whether the vector contains any errors, but then I'd say the for is simpler :-)

Reply:
It is quite simple, just move "?" outside: `v.iter().map(|e| make_where_string(&e.mode)).collect()?`. The magic is that collecting into Result<Vec<T>> will propage the error (it actually stops the iterator on the first error). e.g.: https://stackoverflow.com/questions/26368288/how-do-i-stop-iteration-and-return-an-error-when-iteratormap-returns-a-result

Reply:
Whoa! That is awesome, thanks.